cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java

```diff
@@ -184,6 +193,16 @@ protected boolean needsUpdate(HTableDescriptor tableDescriptor) {
       needUpgrade = true;
     }
 
+
+    String splitsPolicy = spec.getProperty(SPLIT_POLICY);
+    // override using provided split policy
+    if (!Strings.isNullOrEmpty(splitsPolicy) &&
+      (tableDescriptor.getRegionSplitPolicyClassName() == null ||
+        !tableDescriptor.getRegionSplitPolicyClassName().equalsIgnoreCase(splitsPolicy))) {
```

Comment:
You can simplify the condition to

```
if (!Strings.isNullOrEmpty(splitsPolicy) && !splitsPolicy.equals(tableDescriptor.getRegionSplitPolicyClassName()))
```

Also, class name should be compared with `equals`, not `equalsIgnoreCase`.